pkg/apis/serving/v1/revision_defaults.go

```diff
@@ -46,6 +41,11 @@ func (rts *RevisionTemplateSpec) SetDefaults(ctx context.Context) {
 
 // SetDefaults implements apis.Defaultable
 func (rs *RevisionSpec) SetDefaults(ctx context.Context) {
+	// SetDefaults may update revision spec which is immutable.
+	// See: https://github.com/knative/serving/issues/8128 for details.
+	if apis.IsInUpdate(ctx) {
```

Comment:
We had this in Revision because Revision is immutable.  RevisionSpec is not immutable in all contexts.

Putting this here means that pre-existing resources won't have defaulting applied, which seems to me like it'll be breaking.

I'd like to understand what broke a bit better, and if another change is needed, then let's try and do that separately first?

Reply:
Sure.
BAsically here we did upgrade from `nil` EnableServiceLinks to a real value.

Reply:
What do you have in mind that should be defaulted that wasn't before?

Reply:
Note that also those can be defaulted in reconciler itself.